test/SlowTests/Voron/Issues/RavenDB_22973.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using FastTests.Voron;
+using Tests.Infrastructure;
+using Voron;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Voron.Issues;
+
+public class RavenDB_22973 : StorageTest
+{
+    public RavenDB_22973(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    protected override void Configure(StorageEnvironmentOptions options)
+    {
+        base.Configure(options);
+
+        options.ManualFlushing = true;
+        options.ManualSyncing = true;
+    }
+
+    [RavenFact(RavenTestCategory.Voron)]
+    public void ScratchBufferMustNotContainDeletedLeftoversAfterFlushing_AllocateFreeAllocateInSameTx()
```

Comment:
This is the test which is reproducing the issue. Other two tests has been added as precaution.